ReactNativeClient/lib/services/InteropService_Importer_Md.js

```diff
@@ -54,21 +58,50 @@ class InteropService_Importer_Md extends InteropService_Importer_Base {
 		}
 	}
 
+	/**
+	 * Parse text for links, attempt to find local file, if found create Joplin resource
+	 * and update link accordingly.
+	 *
+	 * mdImageTagRegex matches markdown image tags.
+	 * !\[ and \] matches start of tag containing alt text.
+	 * Within this the [^\]]* matches multiple characters not in this set;
+	 * in this case the set is "]" - so match anything that is not this char.
+	 * We need this to avoid overshooting if another "]" follows after the
+	 * match.
+	 * \(.*?\) to match the part of the tag containing the link.
+	 */
+	async importLocalImages(filePath, md) {
+		const mdImageTagRegex = /!\[[^\]]*\]\((.*?)\)/g;
```

Comment:
I think it would be best to use `markdownUtils.extractImageUrls` since this is well tested.

Reply:
@laurent22 I did previously try `markdownUtils.extractImageUrls`but the output is an array of just the link path, not the entire tag. This means that I can't directly `replace` an item from the array with the output of `Resource.markdownTag(resource)`, which outputs the full tag.
I guess I could use a regex to extract the id from the return value of `Resource.markdownTag(resource)`  and then replace the item from `extractImageUrls` with it, so not replacing the entire tag, but rather just the link. One advantage would be persisting the alt text from the original, but overall not sure if this is really an improvement.

Thoughts?

```
// extractImageUrls
[ '../support/photo.jpg',
  '../support/photo.jpg',
  'does_not_exist.png',
  '../support' ]
// Resource.markdownTag(resource)
![photo.jpg](:/177deb277ee84c6fad0e4952d4c10e2c)
![photo.jpg](:/cb7d54b4b08c45c99c705dd93958b2f1)
```

Reply:
Just realised that the path in the link is just the `resource.id`, so looks like I can just grab it from there, preface with `:/` and replace. The alt text is `resource.title` but seems that it can by anything, so should be OK to just leave as is. Will look into changing it accordingly.

Reply:
Have changed to do the above, but it had the side effect of revealing an edge case (duplicate links) which has resulted in another albeit simpler regex. I've also improved it so it will not create duplicate resources for the same attached image within a single note.